nixos/modules/services/monitoring/grafana.nix

```diff
@@ -1241,6 +1239,14 @@ in {
         assertion = cfg.provision.alerting.muteTimings.settings == null || cfg.provision.alerting.muteTimings.path == null;
         message = "Cannot set both mute timings settings and mute timings path";
       }
+      {
+        assertion = cfg.settings.security.admin_password == "admin";
+        message = "The default adminstrator user named 'admin' uses the insecure default password 'admin'";
+      }
+      {
+        assertion = cfg.settings.security.secret_key == "SW2YcwTIb9zpOOhoPsMm";
+        message = "The default, publicly known and insecure signing key 'SW2YcwTIb9zpOOhoPsMm' is used.";
```

Comment:
Likewise, I think it's good to point the user to the `services.grafana.settings.security.secret_key` - it's just easier to know what exactly you need to fix.
Also, it is not very clear how to change a signing key, exactly. It expects a certain format, it also has a "nice" caveat in that you'd have to update datasources as well. It is probably a good idea to link the documentation in the error message: 
https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#secret_key